packages/rocketchat-lib/server/functions/notifications/email.js

```diff
@@ -0,0 +1,169 @@
+}
+
+export function sendEmail({ message, user, subscription, room, emailAddress, toAll }) {
+	let emailSubject;
+	const username = RocketChat.settings.get('UI_Use_Real_Name') ? message.u.name : message.u.username;
+
+	if (room.t === 'd') {
+		emailSubject = RocketChat.placeholders.replace(RocketChat.settings.get('Offline_DM_Email'), {
+			user: username,
+			room: RocketChat.roomTypes.getRoomName(room.t, room)
+		});
+	} else if (toAll) {
+		emailSubject = RocketChat.placeholders.replace(RocketChat.settings.get('Offline_Mention_All_Email'), {
+			user: username,
+			room: RocketChat.roomTypes.getRoomName(room.t, room)
+		});
+	} else {
+		emailSubject = RocketChat.placeholders.replace(RocketChat.settings.get('Offline_Mention_Email'), {
+			user: username,
+			room: RocketChat.roomTypes.getRoomName(room.t, room)
+		});
+	}
+	const content = getEmailContent({
+		message,
+		user,
+		room
+	});
+
+	const link = getMessageLink(room, subscription);
+
+	if (RocketChat.settings.get('Direct_Reply_Enable')) {
+		contentFooter = RocketChat.placeholders.replace(RocketChat.settings.get('Email_Footer_Direct_Reply') || '');
+	}
+
+	const email = {
+		to: emailAddress,
+		subject: emailSubject,
+		html: contentHeader + content + divisorMessage + link + contentFooter
+	};
+
+	// using user full-name/channel name in from address
+	if (room.t === 'd') {
+		email.from = `${ String(message.u.name).replace(/@/g, '%40').replace(/[<>,]/g, '') } <${ RocketChat.settings.get('From_Email') }>`;
+	} else {
+		email.from = `${ String(room.name).replace(/@/g, '%40').replace(/[<>,]/g, '') } <${ RocketChat.settings.get('From_Email') }>`;
+	}
+	// If direct reply enabled, email content with headers
+	if (RocketChat.settings.get('Direct_Reply_Enable')) {
+		email.headers = {
+			// Reply-To header with format "username+messageId@domain"
+			'Reply-To': `${ RocketChat.settings.get('Direct_Reply_Username').split('@')[0].split(RocketChat.settings.get('Direct_Reply_Separator'))[0] }${ RocketChat.settings.get('Direct_Reply_Separator') }${ message._id }@${ RocketChat.settings.get('Direct_Reply_Username').split('@')[1] }`
+		};
+	}
+
+	Meteor.defer(() => {
+		Email.send(email);
+	});
+}
+
+export function shouldNotifyEmail({ disableAllMessageNotifications, statusConnection, emailNotifications, isHighlighted, hasMentionToUser }) {
+
+	// no user or room preference
+	if (emailNotifications == null) {
+		if (disableAllMessageNotifications) {
+			return false;
+		}
+
+		// default server preference is disabled
+		if (RocketChat.settings.get('Accounts_Default_User_Preferences_emailNotificationMode') === 'disabled') {
+			return false;
+		}
+	}
+
+	// user/room preference to nothing
+	if (emailNotifications === 'nothing') {
+		return false;
+	}
+
+	// use connected (don't need to send him an email)
+	if (statusConnection === 'online') {
+		return false;
+	}
+
+	return isHighlighted || emailNotifications === 'all' || hasMentionToUser;
```

Comment:
it is now easy to see that email notifications are not triggered by `@all` mentions. you can see other notifications are testing the `hasMentionToAll`